src/main/java/UtilsService.java

```diff
@@ -0,0 +1,120 @@
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Service
+@Slf4j
+public class UtilsService {
+
+    public boolean isValidJson(String responseText) {
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            mapper.readTree(responseText);
+            return true;
+        } catch (Exception e) {
+            log.error("Invalid JSON response: " + responseText);
+            return false;
+        }
+    }
```

Comment:
The method catches a generic Exception, which can obscure the actual error type. It should catch specific exceptions to provide better error handling and debugging. (Severity: medium)
**Suggestion**: Catch specific exceptions like JsonParseException instead of a generic Exception.